File: im-mobile/lib/models/message_translation.dart
class MessageTranslation {
  final int? id;
  final int messageId;
  final int userId;
  final String sourceLang;
  final String targetLang;
  final String originalContent;
  final String translatedContent;
  final String? provider;
  final String? model;
  final DateTime translatedAt;
  final bool autoTranslated;
  final int? durationMs;

  MessageTranslation({
    this.id,
    required this.messageId,
    required this.userId,
    required this.sourceLang,
    required this.targetLang,
    required this.originalContent,
    required this.translatedContent,
    this.provider,
    this.model,
    required this.translatedAt,
    required this.autoTranslated,
    this.durationMs,
  });

  factory MessageTranslation.fromJson(Map<String, dynamic> json) {
    return MessageTranslation(
      id: json['id'],
      messageId: json['messageId'] ?? 0,
      userId: json['userId'] ?? 0,
      sourceLang: json['sourceLang'] ?? 'en',
      targetLang: json['targetLang'] ?? 'zh-CN',
      originalContent: json['originalContent'] ?? '',
      translatedContent: json['translatedContent'] ?? '',
      provider: json['provider'],
      model: json['model'],
      translatedAt: DateTime.tryParse(json['translatedAt'] ?? '') ?? DateTime.now(),
      autoTranslated: json['autoTranslated'] ?? false,
      durationMs: json['durationMs'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'messageId': messageId,
      'userId': userId,
      'sourceLang': sourceLang,
      'targetLang': targetLang,
      'originalContent': originalContent,
      'translatedContent': translatedContent,
      'provider': provider,
      'model': model,
      'translatedAt': translatedAt.toIso8601String(),
      'autoTranslated': autoTranslated,
      'durationMs': durationMs,
    };
  }
}

class TranslationSettings {
  final int? id;
  final int userId;
  final bool autoTranslate;
  final String preferredTargetLang;
  final String? autoLangWhitelist;
  final String provider;
  final String? apiKey;
  final bool showOriginal;

  TranslationSettings({
    this.id,
    required this.userId,
    this.autoTranslate = false,
    this.preferredTargetLang = 'zh-CN',
    this.autoLangWhitelist,
    this.provider = 'OPENAI',
    this.apiKey,
    this.showOriginal = true,
  });

  factory TranslationSettings.fromJson(Map<String, dynamic> json) {
    return TranslationSettings(
      id: json['id'],
      userId: json['userId'] ?? 0,
      autoTranslate: json['autoTranslate'] ?? false,
      preferredTargetLang: json['preferredTargetLang'] ?? 'zh-CN',
      autoLangWhitelist: json['autoLangWhitelist'],
      provider: json['provider'] ?? 'OPENAI',
      apiKey: json['apiKey'],
      showOriginal: json['showOriginal'] ?? true,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'userId': userId,
      'autoTranslate': autoTranslate,
      'preferredTargetLang': preferredTargetLang,
      'autoLangWhitelist': autoLangWhitelist,
      'provider': provider,
      'apiKey': apiKey,
      'showOriginal': showOriginal,
    };
  }

  TranslationSettings copyWith({
    int? id, int? userId, bool? autoTranslate, String? preferredTargetLang,
    String? autoLangWhitelist, String? provider, String? apiKey, bool? showOriginal,
  }) {
    return TranslationSettings(
      id: id ?? this.id,
      userId: userId ?? this.userId,
      autoTranslate: autoTranslate ?? this.autoTranslate,
      preferredTargetLang: preferredTargetLang ?? this.preferredTargetLang,
      autoLangWhitelist: autoLangWhitelist ?? this.autoLangWhitelist,
      provider: provider ?? this.provider,
      apiKey: apiKey ?? this.apiKey,
      showOriginal: showOriginal ?? this.showOriginal,
    );
  }
}

File: im-desktop/src/types/message-translation.ts
/**
 * 消息翻译类型定义
 */

export interface MessageTranslation {
  id: number;
  messageId: number;
  userId: number;
  sourceLang: string;
  targetLang: string;
  originalContent: string;
  translatedContent: string;
  provider: string;
  model?: string;
  translatedAt: string;
  autoTranslated: boolean;
  durationMs?: number;
}

export interface TranslationRequest {
  messageId: number;
  sourceLang?: string;
  targetLang: string;
  text: string;
  userId: number;
  autoTranslate?: boolean;
}

export interface TranslationSettings {
  id?: number;
  userId: number;
  autoTranslate: boolean;
  preferredTargetLang: string;
  autoLangWhitelist?: string;
  provider: string;
  apiKey?: string;
  showOriginal: boolean;
}

export interface SupportedLanguage {
  code: string;
  name: string;
  nativeName: string;
}

export const SUPPORTED_LANGUAGES: SupportedLanguage[] = [
  { code: 'zh-CN', name: 'Chinese', nativeName: '中文' },
  { code: 'en', name: 'English', nativeName: 'English' },
  { code: 'ja', name: 'Japanese', nativeName: '日本語' },
  { code: 'ko', name: 'Korean', nativeName: '한국어' },
  { code: 'es', name: 'Spanish', nativeName: 'Español' },
  { code: 'fr', name: 'French', nativeName: 'Français' },
  { code: 'de', name: 'German', nativeName: 'Deutsch' },
  { code: 'ru', name: 'Russian', nativeName: 'Русский' },
  { code: 'ar', name: 'Arabic', nativeName: 'العربية' },
  { code: 'pt', name: 'Portuguese', nativeName: 'Português' },
  { code: 'it', name: 'Italian', nativeName: 'Italiano' },
  { code: 'vi', name: 'Vietnamese', nativeName: 'Tiếng Việt' },
  { code: 'th', name: 'Thai', nativeName: 'ไทย' },
  { code: 'id', name: 'Indonesian', nativeName: 'Bahasa Indonesia' },
];
